### PR TITLE
Allow http-api-data 0.5

### PR DESCRIPTION
**Pull request by felixonmars**
Builds fine and all tests pass.

### CODE REVIEW

File: servant-server/servant-server.cabal [hunk 1]
```diff
@@ -73,7 +73,7 @@ library
   -- strict dependency as we re-export 'servant' things.
   build-depends:
       servant             >= 0.19     && < 0.20
-    , http-api-data       >= 0.4.1    && < 0.4.4
+    , http-api-data       >= 0.4.1    && < 0.5.1
```
Comment (endgame) [inline servant-server/servant-server.cabal:76]:
```suggestion
    , http-api-data       >= 0.4.1    && < 0.6
```

File: servant/servant.cabal [hunk 2]
```diff
@@ -92,7 +92,7 @@ library
   -- We depend (heavily) on the API of these packages:
   -- i.e. re-export, or allow using without direct dependency
   build-depends:
-      http-api-data          >= 0.4.1    && < 0.4.4
+      http-api-data          >= 0.4.1    && < 0.5.1
```
Comment (endgame) [inline servant/servant.cabal:95]:
```suggestion
      http-api-data          >= 0.4.1    && < 0.6
```

**Review by endgame — COMMENTED** [review]
Under the PVP, versions `>=0.5 && <0.6` aren't allowed to make breaking changes, so I recommend relaxing the upper bound to `<0.6`.

File: servant-server/servant-server.cabal (hunk 1 above)
Comment (felixonmars) [reply servant-server/servant-server.cabal:76]: Updated.

Comment (endgame) [reply servant-server/servant-server.cabal:76]: It appears you only updated the bounds for `servant` but not `servant-server`.

**Comment by endgame** [comment]
Reposting here because I can't unresolve the conversation above: It looks like the upper bound for `servant-server` is still on `<0.5.1`. Should it be `<0.6`?

*****

As a Hackage Trustee, I have made the following revisions:

* `servant-0.19:` `>=0.4.1 && <0.4.4` -> `>=0.4.1 && <0.6`
* `servant-server-0.19.1`: `>=0.4.1 && <0.4.4` -> `>=0.4.1 && <0.5.1` (I think this could be looser, but I'm revising it to match the PR as it stands.)

**Comment by felixonmars** [comment]
Updated to 0.6 everywhere and resolved conflict.

**Comment by ysangkok** [comment]
Thank you for the PR, a similar change was merged as part of #1663, so I'll close this. Please reopen if there are any issues.

**Pull request closed without merging**